### DISCUSSION TITLE
Attaching a text bar/health bar to sprites

### DISCUSSION THREAD

**Discussion by srgfrog** [bevyengine/bevy#7695] [Q&A] [answered]
Hi, 

I'm writing a multiplayer game where a tank moves around, using bevy_rapier2d for the physics. I'm attaching a text bar with player name and health under the tank, but didn't want the text to rotate when the tank did, so tried the following hierarchy:
```
Player entity (SpatialBundle) with LockedAxes::ROTATION_LOCKED,
     -> Child Tank entity  (SpriteBundle) with Collider::cuboid
         -> Child Turret (SpriteBundle)
     -> Child Name entity Text2dBundle 
```

I use the Velocity::linvel of the player entity to control the velocity and Velocity::angvel of the tank for rotation.

This works for most situations - the text bundle displays correctly, and remains under the tank,when moving and rotating the tank (and turret). However when the tank collides with an object and stops, both the Player entity and the Name keep moving. If I try adding LockedAxes::TRANSLATION_LOCKED to the Tank entity, the tank can move raggedly through objects, following the player entity. 

I saw the bevy_simple_stat_bars plugin, but was hoping there was a more elegant solution using hierarchies. Any suggestions? Thank you.

**Comment by rparrett** [comment, marked as answer]
(Making some assumptions about how the rest of your message was going to go...)

Bevy's transform propagation is all-or-nothing. There's no way to to disable only rotation.

Here are a couple options:
- Write a system that runs after bevy's transform propagation systems that resets the rotation/translation of the text.
- Build your own simple hierarchy (e.g. a `Follows(Entity)` component) and write a system to propagate only the translation from the entity being followed.

Edit: The solution you attempted is interesting, and that sort of thing would generally work, but I'm not sure about the specific issue you're seeing with `bevy_rapier`.

**Reply by srgfrog** [reply]
I wrote a quick function to fix this situation. I guess it comes down to I expected that with the following structure 
```
Parent
     -> Child with Collider::cuboid and LockedAxes::TRANSLATION_LOCKED
```
I was expecting that if the Child is unable to move in a direction because of the Collider, and that the parent/child relative translation is locked, then the parent should not be able to move either, but this is not the case. Here is a quick workaround for my player names drifting away from the tank:
```
fn correct_player_tank_position(
    mut tank_info: Query<&mut Transform, Without<Player>>,
    mut player_info: Query<(&mut Transform, &Children), With<Player>>,
) {
    for (mut player_trans, children) in player_info.iter_mut() {
        let tank_child = children.first().expect("No first child");
        if let Ok(mut tank_trans) = tank_info.get_mut(*tank_child) {
            if tank_trans.translation[0] != 0. || tank_trans.translation[1] != 0. {
                player_trans.translation += tank_trans.translation;
                tank_trans.translation = Vec3::ZERO;
            }
        }
    }
}
```
